docs/bookstore/page-2/sentiment-analysis-service-for-bookstore-reviews.md

```diff
@@ -0,0 +1,414 @@
+# Sentiment Analysis Service for Bookstore Reviews
+
+![Image1](images/image1.png)
+
+As a bookstore owner, you aim to receive instant notifications in a Slack channel whenever a customer submits a new negative review comment. By leveraging Knative Function, you can set up a serverless function that contains a simple sentiment analysis service to categorize review comments by sentiment.
+
+## **Which Knative features will we learn about?**
+
+The ease to use **Knative Function** to deploy your service, and make it be managed by Knative Serving, which gives you the ability to **auto-scale your service to zero**, and scale up to handle the demand.
+
+## **What does the final deliverable look like?**
+
+![Image8](images/image8.png)
+
+A running serverless Knative Function that contains a python application that receives the new review comments as CloudEvent and returns the sentiment classification of the input text as CloudEvent.
+
+The function's output will be only from:
+
+- Positive
+
+- Neutral
+
+- Negative
+
+## **Install Prerequisites**
+
+### **Prerequisite 1: Install Knative func CLI**
+
+![Image12](images/image12.png)
+
+Knative Function enables you to easily create, build, and deploy stateless, event-driven functions as [Knative Services](https://knative.dev/docs/serving/services/#:~:text=Knative%20Services%20are%20used%20to,the%20Service%20to%20be%20configured){:target="_blank"} by using the func CLI.
+
+In order to do so, you need to install the func CLI. You can follow the [official documentation](https://knative.dev/docs/getting-started/install-func/){:target="_blank"} to install the func CLI.
+
+???+ success "Verify"
+    Running `func version` in your terminal to verify the installation, and you should see the version of the func CLI you installed.
+
+???+ bug "Troubleshooting"
+    If you see `command not found`, you may need to add the func CLI to your PATH.
+
+## **Implementation**
+
+![Image3](images/image3.png)
+
+The process is straightforward:
+
+1. Begin by utilizing the `func create` command to generate your code template.
+
+2. Next, incorporate your unique code into this template.
+
+3. Finally, execute `func deploy` to deploy your application seamlessly to the Kubernetes cluster.
+
+This workflow ensures a smooth transition from development to deployment within the Knative Functions ecosystem.
+
+### **Step 1: Create a Knative Function template**
+
+![Image17](images/image17.png)
+
+Create a new function using the func CLI:
+
+```
+func create -l <language> <function-name>
+```
+
+In this case, we are creating a Python function, so the command will be:
+
+```
+func create -l python sentiment-analysis-app
```

Comment:
At some ppint we need to update this.

We are currently rewriting the py middleware, and will likely create also a few new templates. 
But that is second half of 2024 - hopefully /cc @lkingland

Reply:
Issue created to keep track this. https://github.com/knative/docs/issues/6044